### PR TITLE
Better error message when options is not specified

### LINKED ISSUE #903: Dropdown throws confusing error when options is not specified
This apparently used to work (it's in the alignment example notebook):

```python
from ipywidgets import Dropdown
Dropdown(description='Foobar')
```

It doesn't now, but the error is very confusing. We should make a validator if we want to insist that the options list is nonempty.

### PR DESCRIPTION
**Pull request by ivanov**
closes #903

### CODE REVIEW

File: ipywidgets/widgets/widget_selection.py [hunk 1]
```diff
@@ -15,21 +15,23 @@
 from .valuewidget import ValueWidget
 from .widget import register
 from traitlets import (Unicode, Bool, Any, Dict, TraitError, CaselessStrEnum,
-                       Tuple, List, Union, observe, validate)
+                       Tuple, List, Union, observe, default, validate)
```
Comment (jasongrout) [inline ipywidgets/widgets/widget_selection.py]: We don't need this import, right?

Comment (ivanov) [inline ipywidgets/widgets/widget_selection.py]: good catch - fat fingered my `git add -p`. Fixed now

File: ipywidgets/widgets/widget_selection.py [hunk 2]
```diff
@@ -27,9 +27,11 @@ def _value_to_label(value, obj):
     # TODO: make a separate validation function so this doesn't have
     # to redo the work of parsing the options object.
     options = obj._make_options(obj.options)
+    if len(options) == 0:
+        raise TraitError("`options` must not be empty.")
     try:
-        # use an iterator approach as a shortcut
-        return next(k for (k, v) in options if obj.equals(v, value))
+        # return the first label whose value is equal the desired value
```
Comment (jasongrout) [inline ipywidgets/widgets/widget_selection.py]: Sorry, one more thing: "equal *to* the desired value"

Comment (ivanov) [inline ipywidgets/widgets/widget_selection.py]: Not at all, thank you! Fixed the typo, now.

**Comment by jasongrout** [comment]
Thanks!

**Pull request merged**